modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/TableScanNode.java

```diff
@@ -241,9 +241,7 @@ public void onNext(BinaryRow binRow) {
 
             inBuff.add(row);
 
-            if (++received == inBufSize) {
-                received = 0;
-
+            if (inBuff.size() == inBufSize) {
```

Comment:
```suggestion
            if (inBuff.remainingCapacity() == 0) {
```

Reply:
I would suggest rewrite all such place to use remainingCapacity(), it's more clear.

Reply:
In such a case we need to specify queue implementation i don`t think it`s good idea.

Reply:
it could be just `BlockingQueue` interface

Reply:
we don`t use BlockingQueue features here, or i miss smth ? It can be easily changed in future.

Reply:
I suppose no need for using BlockingQueue here, it`s by now only for convenience